gemrb/core/Game.cpp

```diff
@@ -1570,8 +1570,18 @@ void Game::UpdateScripts()
 		// perhaps a StartMusic action stopped the area music?
 		// (we should probably find a less silly way to handle this,
 		// because nothing can ever stop area music now..)
-		doChangeSong = true;
+
+		// Call ChangeSong only once per round in order to prevent spam
+		// calls every tick in areas without music.
+		static int ticks = 0;
```

Comment:
The music plays when paused, so we can't just use GameTime here. However, if the game is paused, no interaction can occur, so nothing would change the song (at most it could run out if not looping). So you actually can use GameTime.

Reply:
Looks like the originals tolerated 10 ai updates without music, before rechecking:
https://github.com/alexbatalov/iwd2-re/blob/4fa36bcd6c33e0d35c9e273561aa683aece88490/src/CGameArea.cpp#L1309

Reply:
I don't fully understand. Do we need to bring GameTime into the mix or is it ok to replace the tick >= round comparision with a tick >= 10 comparison?

Reply:
I'm saying you don't need an extra timer. Looking at the code again, yeah, 10 is the default instead of round size (75/90/105), but for day/night song it only checks every 600. And an extra complication for combat. So I think we can just simplify it and leave it at round size intervals.